webapp/components/sidebar.jsx

```diff
@@ -382,11 +383,13 @@ export default class Sidebar extends React.Component {
     }
 
     sortChannelsByDisplayName(a, b) {
+        const locale = LocalizationStore.getLocale();
+
         if (a.display_name === b.display_name) {
-            return a.name.localeCompare(b.name);
+            return a.name.localeCompare(b.name, locale, {numeric: true, caseFirst: 'upper'});
```

Comment:
But if we leave it blank it will choose the sort order based on locale so people will be used to that sort order presumably. It really does make sense to leave that at default.